### PR TITLE
Task 181/z index implementation

### PR DESCRIPTION
**Pull request by codeFactory12**
# US Related

[Task](https://tree.taiga.io/project/joseluis-teran-coffeetime/task/181)

## Description

Implementation of z-index, for the priority of order at the momment of graph.
- #### What did I do?
- Implemented z index in Coords, edit method Add of Repository, create method update in order the game developer can update in the logic of the game the priority of the Sprites 
- #### How did I do it?
- Editig and Adding classes, also making tests about it
- #### Why did I do it?
- In order to have a order for the presentation of Sprites

## Type of Change

Put an `x` in all the boxes that apply:

- [x] ✨ New Feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug Fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking Change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code Refactor
- [ ] ✅ Build Configuration Change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

## Pull Request Checklist

Put an `x` in all the boxes that apply:

- [x] My commit messages are detailed.
- [ ] My code follows the code style of this project.
- [ ] No existing features have been broken without good reason.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Review by zrock-dev — CHANGES_REQUESTED** [review]
I like the idea of having SpriteRepository order the sprites based on Z. I think that Coord is meant to express the position of the sprite in the screen, so Z would be understood as a spacial position instead of ordinal. What about adding Z as an attribute of Sprite?

File: bridge/src/main/java/com/bridge/renderHandler/repository/SpriteRepository.java [hunk 1]
```diff
@@ -21,7 +22,14 @@ public SpriteRepository() {
      */
     @Override
     public void add(Sprite sprite) {
-        sprites.add(sprite);
+        sprites.add(sprite.getZ_index(), sprite);
```
Comment (zrock-dev) [inline bridge/src/main/java/com/bridge/renderHandler/repository/SpriteRepository.java]: >  Shifts the element currently at that position (if any) and any subsequent elements to the right (adds one to their indices).
-- [source](https://docs.oracle.com/en/java/javase/17/docs//api/java.base/java/util/List.html#add(int,E))

I think this add method would shift sprites from their layers. What about using the retrieve method instead, to return a list ordered by its z index?

**Review by zrock-dev — CHANGES_REQUESTED** [review]
Nice, our coverage has been increased.

File: bridge/src/main/java/com/bridge/renderHandler/repository/SpriteRepository.java (hunk 1 above)
Comment (zrock-dev) [reply bridge/src/main/java/com/bridge/renderHandler/repository/SpriteRepository.java]: What I meant was to add sprites to the list without the specific index.
`spites.add(sprite)`

**Comment by zrock-dev** [comment]
this is a erply test from email

**Review by zrock-dev — APPROVED** [review]
Alrighty, nicely done

**Review by victor-villca — APPROVED** [review]
Good, i like the way to manage the zindex.

**Review by fabianromeroclaros — APPROVED** [review]

**Pull request merged**